src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,156 @@
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def delete(self, request, organization, member):
+        audit_data = member.get_audit_log_data()
+        if OrganizationMemberDetailsEndpoint._is_only_owner(member):
+            return Response({"detail": ERR_ONLY_OWNER}, status=403)
+        with transaction.atomic():
+            AuthIdentity.objects.filter(
+                user=member.user, auth_provider__organization=organization
+            ).delete()
+            member.delete()
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=member.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_REMOVE,
+                data=audit_data,
+            )
+
+        return Response(status=204)
+
+
+class OrganizationSCIMUserIndex(SCIMEndpoint):
+    def get(self, request, organization):
+        # note that SCIM doesn't care about changing results as they're queried
+        # TODO: sanitize get parameter inputs?
+
+        filter_val = parse_filter_conditions(request.GET.get("filter"))
+
+        queryset = (
+            OrganizationMember.objects.filter(
+                Q(invite_status=InviteStatus.APPROVED.value),
+                Q(user__is_active=True) | Q(user__isnull=True),
+                organization=organization,
+            )
+            .select_related("user")
+            .order_by("email", "user__email")
+        )
+        if filter_val:
+            queryset = queryset.filter(
+                Q(email__in=filter_val) | Q(user__email__in=filter_val)
+            )  # not including secondary email vals (dups, etc.)
+
+        # TODO: how is queryset ordered?
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        return self.paginate(
+            request=request,
+            on_results=lambda results: serialize(results, None, OrganizationMemberSCIMSerializer()),
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+        )
+
+    def post(self, request, organization):
+        # TODO: confirm mixed case emails get converted to lowercase
+        serializer = OrganizationMemberSerializer(
+            data={
+                "email": request.data.get("userName"),
+                "role": roles.get(organization.default_role).id,
+            },
+            context={
+                "organization": organization,
+                "allowed_roles": [roles.get(organization.default_role)],
+                "allow_existing_invite_request": True,
+            },
+        )
+
+        if not serializer.is_valid():
+            # TODO: other ways this could be invalid?
+            return Response(SCIM_409_USER_EXISTS, status=409)
```

Comment:
Could the serializer errors be used?

Reply:
yes -- albiet have to do a bit of a hack (parsing the string of the error) so we can have it return a 409 instead of a standard 400. best solution i see without refactoring / duplicating code.